### DISCUSSION TITLE
How to use SET file in MT5?

### DISCUSSION THREAD

**Discussion by Harry040** [EA31337/EA31337#332] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/12960656/152088436-fca06c8b-f5a1-4c8b-85bf-04c81474f576.png)
I have some questions about the file

1. how to generate the set file?
2. how to use the set file?

**Comment by kenorb** [comment, marked as answer]
> How to generate the SET file in MT4

You can generate it either manually by editing the file or through the manual export (from the MetaTrader platform) by selecting the option you want to optimize.

When you're editing the file manually, then:

- line with 1 is the starting value to optimize
- line with 2 is the step value (0 for enums)
- line with 3 is the stop value (inclusive)
- line with F enables optimization for the given param (1 to enable)

So in the above example, you have optimization between -40 and 0 with step 5 (9 iterations).

> How to use the SET file for MT4?

You can load it either:

- manually in MT4 platform,
- when using [EA Tester](https://github.com/EA31337/EA-Tester), use  `-f (file)` and `-o` to run in optimization mode (see: [`options.txt`](https://github.com/EA31337/EA-Tester/blob/v1.0.3/scripts/options.txt))
- when using [MQL-Tester-Action](https://github.com/FX31337/MQL-Tester-Action), you use `SetFile` input.


> How to generate the SET file in MT5

For MetaTrader 5, SET files have different formats and they're not supported by the EA Tester. They need to be converted or exported manually. Then you've to use manually run the optimize process in the MT5 platform.

**Reply by Harry040** [reply]
Appreciate for the comprehensive replay!
